### ISSUE TITLE
Questions on using std::streambuf

### ISSUE THREAD

**Issue by drorgy** [Azure/azure-storage-cpp#363]
Hi, 

I already have a std::streambuf implementation (my underlying data is a complex multi array structure). Is there a way I can wrap it over an ostream and use it to download a blob? (and same question or upload).

For example: 
Where I would normally do:
concurrency::streams::container_buffer<std::vector<uint8_t>> buffer;
concurrency::streams::ostream output_stream(buffer);
blockBlob.download_to_stream(output_stream);
I want to do:
MyStdStreamBufDerivedClass buffer;
??::??::ostream output_stream(buffer);
blockBlob.download_to_stream(output_stream);

Possibly if there is an adapter mechanism between an STL streamBuf/ostream/istream to the Casablanca counterparts - that would also be good.

Thanks,
Dror

**Comment by JinmingHu-MSFT** [comment]
@drorgy Yes, I believe there's a way to download to (or upload from) a multi-array structure. But that can be kind of complicated. You need to inherit from something in `concurrency::stream`, which is part of [cpprestsdk](https://github.com/microsoft/cpprestsdk). I recommend you file an issue in that repo. Maintainers of that project are more familiar with these stream stuff than we do.

**Comment by JinmingHu-MSFT** [comment]
@drorgy I noticed you're trying out both this cpp sdk and cpplite sdk. We're currently actively working on [Track2 C++ SDK](https://github.com/Azure/azure-sdk-for-cpp/tree/master/sdk/storage), which in the future will replace both cpp and cpplite sdk.
